test/functional/bootstrap/FeatureContext.php

```diff
@@ -166,11 +167,11 @@ public function iAddObjectsOfTypeToIndexWithData($objectType, $indexName, TableN
         $objectIndexer = new ObjectIndexer($this->getIndex($indexName));
 
         $objectListHash = $objectList->getHash();
+        $indexableObjects = [];
         foreach ($objectListHash as $objectListRow) {
-            $indexableObject = new IndexableObject($objectListRow['id'], $objectListRow);
-            $objectIndexer->index($indexableObject, $objectType);
+            $indexableObjects[] = new IndexableObject($objectListRow['id'], $objectListRow);
         }
-
+        $objectIndexer->bulkIndex($indexableObjects, $objectType);
```

Comment:
That bothers me because we then lose coverage for the single index method.

Maybe something like an `array_pop` to extract 1 element and index it with the old method will do the trick.

Reply:
I took care to check that it was used elsewhere :+1: 
```
    public function iUpdateObjectOfTypeWithIdInIndexWithData($objectType, $id, $indexName, TableNode $objectList)
    {
        $objectIndexer = new ObjectIndexer($this->getIndex($indexName));
        $newData = $objectList->getHash()[0];
        $objectIndexer->index(new IndexableObject($id, $newData), $objectType);
    }
``` 
```
    public function iCreateObjectsOfTypeMyGeoTypeToIndex(string $objectType, string $indexName)
    {
        $objectIndexer = new ObjectIndexer($this->getIndex($indexName));

        $cityArray =
            [
                ['id' => 1, 'city_name' => 'lyon', 'location' => ['lat' => '45.764043', 'lon' => '4.835658999999964' ]],
                ['id' => 2, 'city_name' => 'paris', 'location' => ['lat' => '48.85661400000001', 'lon' => '2.3522219000000177' ]],
                ['id' => 3, 'city_name' => 'mâcon', 'location' => ['lat' => '46.30688389999999', 'lon' => '4.828731000000062' ]]
            ];

        foreach ($cityArray as $cityRow) {
            $indexableObject = new IndexableObject($cityRow['id'], $cityRow);
            $objectIndexer->index($indexableObject, $objectType);
        }

        sleep(1);
    }
```
```

    /**
     * @When I create nested index and populate it on :indexName
     */
    public function iCreateObjectsOfNestedTypeToIndex($indexName)
    {
        $objectIndexer = new ObjectIndexer($this->getIndex($indexName));

        $cityArray =
            [
                ['id' => 1, 'title' => 'Incredible Hulk', 'authors' => [
                    ['first_name' => 'Jack', 'last_name' => 'Kirby'] ,
                    ['first_name' => 'Stan', 'last_name' => 'Lee' ]
                ]],
            ];

        foreach ($cityArray as $cityRow) {
            $indexableObject = new IndexableObject($cityRow['id'], $cityRow);
            $objectIndexer->index($indexableObject, '_doc');
        }

        sleep(1);
    }
```

and thought it was enough

Reply:
It is. I missed it.